src/App.tsx

```diff
@@ -1,223 +1,155 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import React, {
+  useCallback,
+  useContext, useEffect, useMemo, useRef, useState,
+} from 'react';
 import { AuthContext } from './components/Auth/AuthContext';
+import { createTodo, deleteTodo, getTodos } from './api/todos';
+import { Todo } from './types/Todo';
+import {
+  ErrorNotification,
+} from './components/ErrorNotification/ErrorNotification';
+import { TodoList } from './components/TodoList/TodoList';
+import { Header } from './components/Header/Header';
+import { Footer } from './components/Footer/Footer';
+import { FilterType } from './types/FilterType';
+import { filteredTodosHelper } from './helpers/FilteredTodos';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [completedFilter, setCompletedFilter] = useState(FilterType.All);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [isAdding, setIsAdding] = useState(true);
+
   const user = useContext(AuthContext);
   const newTodoField = useRef<HTMLInputElement>(null);
 
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
-    if (newTodoField.current) {
-      newTodoField.current.focus();
+    if (user) {
+      getTodos(user?.id)
+        .then(setTodos)
+        .catch(() => {
+          setHasError(true);
+          setErrorMessage('Can\'t load todos');
+        });
+    }
+  });
+
+  const filteredTodos = useMemo(() => (
+    filteredTodosHelper(todos, completedFilter)
+  ), [todos, completedFilter]);
+
+  const countOfActiveTodos = useMemo(() => {
+    return todos.filter(todo => !todo.completed).length;
+  }, [todos]);
+
+  const hasCompletedTodos = useMemo(() => {
+    return todos.some(todo => todo.completed);
+  }, [todos]);
+
+  if (hasError) {
+    setTimeout(() => setHasError(false), 3000);
+  }
+
+  const closeMessage = () => {
+    setHasError(false);
+  };
+
+  const addNewTodo = useCallback(async (newTitle: string) => {
+    if (!newTitle.trim()) {
+      setHasError(true);
+      setErrorMessage('Title can\'t be empty');
+
+      return;
+    }
```

Comment:
Better to move validation to form component

Reply:
**Fixed**